robot/impl/resource_manager.go

```diff
@@ -629,6 +629,41 @@ func (manager *resourceManager) completeConfig(
 	levels := manager.resources.ReverseTopologicalSortInLevels()
 	timeout := rutils.GetResourceConfigurationTimeout(manager.logger)
 	for _, resourceNames := range levels {
+		// if any resources in the level needs reconfiguration and depends on weak dependents,
```

Comment:
```suggestion
		// if any resources in the level need reconfiguration and have weak dependencies,
```
The phrase "depend on weak dependents" is pretty confusing to me. Is my rephrasing there accurate with what you're meaning?

Reply:
it's not accurate - what I mean is if any resource in the level need reconfiguration and depends on a resource that have weak dependencies. So if the resource depends on something that would be affected by updateWeakDependents.

Reply:
Thanks for clarifying; I see now you're using `GetAllParentsOf` on resources in the level and then checking whether those parents (resources that depend on the original resource) has weak dependencies. Now I'm more confused on how we arrived on that being part of the requirement for `updateWeakDependents`; let's talk about in the meeting later.